services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala

```diff
@@ -17,23 +17,55 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
   private val configurationService = new ProjectConfigurationService(configurationRepository)
 
   "ConfigurationServiceTest" when {
+    val projectFileConfiguration =
+      ProjectFileConfiguration(Paths.get("/home/file"), Set(FileParameter("nodeName", StringTyped(Some("hello")))))
     val configuration = ProjectConfiguration(
       TestProjectUtils.getDummyProjectId,
       active = true,
-      List(
-        ProjectFileConfiguration(Paths.get("/home/file"), List(FileParameter("nodeName", StringTyped(Some("hello")))))
-      )
+      Set(projectFileConfiguration)
     )
     val activeDocument = ProjectConfiguration.toDocument(configuration)
     val inactiveDocument = ProjectConfiguration.toDocument(configuration.copy(active = false))
     val updateResult = mock[UpdateResult]
 
     "add configuration" should {
-      "return complete status for creating configuration" in {
+      val projectId = configuration.projectId.value
+
+      "return complete status for updating exist configuration" in {
+        val newProjectFileConfiguration =
+          projectFileConfiguration.copy(inputs = Set(FileParameter("nodeName", StringTyped(Some("hi")))))
+        val newConfiguration = configuration.copy(projectFileConfigurations = Set(newProjectFileConfiguration))
+        val newActiveDocument = ProjectConfiguration.toDocument(newConfiguration)
+
+        when(configurationRepository.getByParam("projectId", projectId))
+          .thenReturn(Future.successful(List(activeDocument)))
+        when(updateResult.toString).thenReturn("Success update")
+        when(configurationRepository.updateOne(newActiveDocument, "projectId", projectId))
+          .thenReturn(Future.successful(updateResult))
+
+        configurationService.addConfiguration(newConfiguration).map(_ shouldBe "Success update")
+      }
+
+      "return complete status for adding new configuration" in {
```

Comment:
I wonder why you needed to change an existing test.
If you did everything correctly, then you shouldn't change existing test for configuration creation. Why is it changed?

Reply:
That's because the old test just replaces the configuration without checking if configuration with the same path exists.

Reply:
I get it, but it should not affect test implementation that much. You should just add one more mocked call for `getByParam` method and it should be enough.

Reply:
Fixed.